### DISCUSSION TITLE
Automatic Scheduled Backup System

### DISCUSSION THREAD

**Discussion by MaXtorBE** [tillsteinbach/VWsFriend#651] [Ideas]
I had an issue with my docker installation recently and had to hard reset it.  I was not able to get the dbase or make a backup of it before proceding with the docker repair...  My last backup that I took was 4 months old...  So lost a lot of charging and driving data...  while this is not life ending, it messes up my overall statistics :(

That's why I would like to suggest in VWsFriend an automatic backup system where we can specify:
- Time interval (daily/weekly/monthly) (ma/tu/we/th/fr/sa/su) (jan/feb/mar/apr/may/jun/jul/aug/sep/oct/nov/dec)
- Backup location
- Retention span (last [x] backups / keep always)

If this is possible, I and other users will never loose data again.

**Comment by tillsteinbach** [comment]
The nice thing is, with docker you already have the tools in hand. Have a look here: https://offen.github.io/docker-volume-backup
You just add some config and can backup wherever you want, with compression, retention, ...
I might add an example to the Readme

**Comment by MaXtorBE** [comment]
The problem is, I’m just knowledgeable enough to get your software running in docker…. And that was a miracle on its own. Having to do this with something else scares the 💩 out of me.Send from my iPhoneOp 24 aug 2024 om 00:00 heeft Till Steinbach ***@***.***> het volgende geschreven:﻿
The nice thing is, with docker you already have the tools in hand. Have a look here: https://offen.github.io/docker-volume-backup
You just add some config and can backup wherever you want, with compression, retention, ...
I might add an example to the Readme
